### PR TITLE
Revert using a separate draw thread

### PR DESCRIPTION
**Pull request by andydotxyz**
Progresses #4654 

### Checklist:

- [ ] Tests included. <- currently manual testing to check all platforms
  - [x] macOS arm
  - [x] macOS Intel
  - [x] Windows
  - [x] Linux X11
  - [x] Linux Wayland
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.
- [x] Any breaking changes have a deprecation path or have been discussed.

Notice that on macOS the UI will stop drawing if we resize. This is the reason draw thread was added and we will need to find another solution...

### CODE REVIEW

File: internal/animation/runner.go [hunk 1]
```diff
@@ -61,32 +60,34 @@ func (r *Runner) Stop(a *fyne.Animation) {
...
+	oldList := r.animations
+	r.animationMutex.Unlock()
+	newList := make([]*anim, 0, len(oldList))
+	for _, a := range oldList {
+		if !a.isStopped() && r.tickAnimation(a) {
+			newList = append(newList, a)
 		}
+	}
+	r.animationMutex.Lock()
+	r.animations = append(newList, r.pendingAnimations...)
```
Comment (dweymouth) [inline internal/animation/runner.go:81]: The runner can get a whole lot cleaner once we assume the new threading model. While Start and Stop can still be called when invoking user code within a Tick, we will know they won't be called elsewhere, so we can do some simplification and cleanup around the synchronization between Start, Stop and Tick. That can be a follow-up PR though.

**Comment by andydotxyz** [comment]
Putting this on hold until we begin work on 2.6

**Comment by andydotxyz** [comment]
Reworked on top of `develop`.

This seems to work smoothly in limited testing but we need to get it checked on all OS.
Lots more work is required to simplify but I didn't want to go too deep yet.

**Comment by andydotxyz** [comment]
P.S. in a follow up PR where I merge the event thread in as well I have **hello** and **fyne_demo** completely without race errors!

Keeping them separate as this PR works smoothly but the event management change has significant performance impact!

**Comment by andydotxyz** [comment]
All Mac tests passing. There is a performance impact but it is actually clear where some latency is coming from - hovering events are triggering multiple event updates etc. This change may help with hunting down other performance bugs!

**Comment by Jacalz** [comment]
Nice. That should help us decrease power consumption and unnecessary cpu usage :)

File: internal/driver/glfw/loop.go [hunk 2]
```diff
@@ -55,16 +48,8 @@ func runOnMain(f func()) {
...
-		runOnMain(func() { w.RunWithContext(f) })
-		return
-	}
-	done := common.DonePool.Get()
-	defer common.DonePool.Put(done)
-
-	drawFuncQueue <- drawData{f: f, win: w, done: done}
-	<-done
+func runOnMainWithContext(w *window, f func()) {
+	runOnMain(func() { w.RunWithContext(f) }) // TODO remove this completely
```
Comment (Jacalz) [inline internal/driver/glfw/loop.go:52]: Is this a TODO for an upcoming PR or should it be removed?

**Review by Jacalz — APPROVED** [review]
Well done. Performance seems okay and everything appears to be working here on Linux. Just left one minor comment line but I'll approve before hand :)

File: internal/driver/glfw/loop.go (hunk 2 above)
Comment (andydotxyz) [reply internal/driver/glfw/loop.go:52]: Yes it is part of the future refactoring. Lots more to be done as mentioned above and in the other tickets.

There are 3 different ways to queue onto what is essentially the same thread but I didn't want to rip out too much code until the direction is established :)

Comment (Jacalz) [reply internal/driver/glfw/loop.go:52]: Cool. Just wanted to check. Still approved from my end :)

**Pull request merged**